src/Utils/AttributesUtils.php

```diff
@@ -0,0 +1,145 @@
+			return null;
+		};
+
+		$has_named_font_size  = ! empty( $font_size );
+		$has_custom_font_size = isset( $custom_font_size );
+
+		if ( $has_named_font_size ) {
+			return array(
+				'class' => sprintf( 'has-font-size has-%s-font-size', $font_size ),
+				'style' => null,
+			);
+		} elseif ( $has_custom_font_size ) {
+			return array(
+				'class' => null,
+				'style' => sprintf( 'font-size: %s;', $custom_font_size ),
+			);
+		}
+
+		return array(
+			'class' => null,
+			'style' => null,
+		);
+	}
+
+	/**
+	 * Get class and style for text-color from attributes.
+	 *
+	 * @param array $attributes Block attributes.
+	 *
+	 * @return (array | null)
+	 */
+	public static function get_text_color_class_and_style( $attributes ) {
+
+		$text_color        = $attributes['textColor'];
+		$custom_text_color = $attributes['style']['color']['text'];
+
+		if ( ! isset( $text_color ) && ! isset( $custom_text_color ) ) {
+			return null;
+		};
+
+		$text_color_class = sprintf( 'has-text-color' );
+		$text_color_style = sprintf( 'color: %s;', ( ! empty( $text_color ) ? $text_color : $custom_text_color ) );
+
+		return array(
+			'class' => $text_color_class,
+			'style' => $text_color_style,
+		);
+	}
+
+	/**
+	 * Get class and style for link-color from attributes.
+	 *
+	 * @param array $attributes Block attributes.
+	 *
+	 * @return (array | null)
+	 */
+	public static function get_link_color_class_and_style( $attributes ) {
+
+		$link_color = $attributes['style']['elements']['link']['color']['text'];
+
+		if ( ! isset( $link_color ) ) {
+
+			return null;
+		};
+
+		$link_color_class = sprintf( 'has-link-color', $link_color );
+		$link_color_style = sprintf( 'color: %s;', $link_color );
+
+		return array(
+			'class' => $link_color_class,
+			'style' => $link_color_style,
+		);
+	}
+
+	/**
+	 * Get class and style for line height from attributes.
+	 *
+	 * @param array $attributes Block attributes.
+	 *
+	 * @return (array | null)
+	 */
+	public static function get_line_height_class_and_style( $attributes ) {
+
+		$line_height = $attributes['style']['typography']['lineHeight'];
+
+		if ( ! isset( $line_height ) ) {
+			return null;
+		};
+
+		$line_height_style = 'line-height: ' . $line_height . ';';
+
+		return array(
+			'class' => null,
+			'style' => $line_height_style,
+		);
+	}
+
+	/**
+	 * Get classes and styles from attributes.
+	 *
+	 * @param array $attributes Block attributes.
+	 *
+	 * @return array
+	 */
+	public static function get_classes_and_styles_by_attributes( $attributes ) {
+		$classes_and_styles = array(
+			line_height => self::get_line_height_class_and_style( $attributes ),
+			text_color  => self::get_text_color_class_and_style( $attributes ),
+			font_size   => self::get_font_size_class_and_style( $attributes ),
+			link_color  => self::get_link_color_class_and_style( $attributes ),
+		);
+
+		return array_filter(
+			$classes_and_styles,
+			function ( $var ) {
+				return ! is_null( $var );
+			}
+		);
```

Comment:
`array_filter` default callbackup already removes all falsy items, so we can use it without the callback.
```suggestion
		return array_filter( $classes_and_styles );
```